pkg/splunk/enterprise/configuration.go

```diff
@@ -472,15 +474,32 @@ func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaOb
 	return statefulSet, nil
 }
 
+// getAppListingConfigMap returns the App listing configMap, if it exists and applicable for that instanceType
+func getAppListingConfigMap(client splcommon.ControllerClient, cr splcommon.MetaObject, instanceType InstanceType) (*corev1.ConfigMap, bool) {
+
+	// ToDo: sgontla: Exclude MC, once it's own CR is available
+	if instanceType != SplunkIndexer && instanceType != SplunkSearchHead && instanceType != SplunkMonitoringConsole {
```

Comment:
Why is it only applicable for CM and Deployer?

Reply:
Applicable to any pod other than the one we are checking here.

Reply:
Yeah that was the question, why not applicable to other pods? :)

Reply:
AS you see in the above conditional check If the pod type is Indexer/SH, the configuration is applied through the CM/Deployer, so, there is nothing that we need to propagate in those cases. Once, we have a separate CR for the Deployer, ideally, we don't even need this check. Anyhow, it is a good one-time check before finalizing the pod template.